### PR TITLE
Add all project files

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    with:
+      repo-name: dr2-ip-lock-checker
+      artifact-name: ip-lock-checker
+      build-command: |
+        mkdir package
+        pip install --target ./package -r requirements-runtime.txt
+        cd package
+        zip -r ../ip_block_checker_lambda.zip .
+        cd ..
+        zip notifications_lambda.zip lambda_function.py
```
Comment (MancunianSam) [inline .github/workflows/build.yml:21]: You need `ip_block_checker_lambda.zip` on this line.

File: lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        if any_other_website_received_unexpected_response:
+            print("Preservica website timed out as expected but other test websites did not receive expected response.")
+            return rest_of_the_tested_websites
+        return {}
+    else:
+        return {preservica_website.name: preservica_website}
+
+
+def lambda_handler(verify_responses_func = verify_responses, print_error_message_func=print_error_message,
```
Comment (MancunianSam) [inline lambda_function.py:97]: For the lambda to work properly, you need a function which takes (event, context) for the lambda to call.  
So here, if you rename this to `run_connection_tests` or something and then have:
```python
def lambda_handler(event, context):
    run_connection_tests()
```

Once that's in, you can get rid of the 
```python
if __name__ == "__main__":
    lambda_handler()
```
as you shouldn't need it.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Load all runtime and dev packages
```
Comment (MancunianSam) [inline requirements.txt:1]: You could get rid of this and rename `requirements-runtime` to `requirements` as you're not aggregating any requirements file. It's up to you though, if you want to leave it so it matches the others.

File: lambda_function.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+def get_response(request, url: str) -> int | Exception:
+    try:
+        response: BaseHTTPResponse = request("GET", url, timeout = urllib3.Timeout(5), retries = False)
+        return response.status
+    except Exception as e:
+        return e
+
+
+def verify_responses(website_to_test_expected_responses: dict[str, Website],
+                     request = http.request) -> dict[str, Website]:
```
[the remark this answers is not in the document]
Comment (techncl) [reply lambda_function.py:35]: Weirdly, my Pycharm adds spaces around keyword arguments; I'll switch it off.

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**